test/queue.js

```diff
@@ -0,0 +1,90 @@
+import chai, { expect } from 'chai'
+import chaiChange from 'chai-change'
+import Queue from '../src/queue'
+
+chai.use(chaiChange)
+  let queue
+
+describe('Queue', function () {
+  beforeEach(function(){
+    queue = new Queue()
+  })
+
+  it('exists', function () {
+    expect(Queue).to.be.a('function')
+  })
+
+  context('enqueue()', function () {
+
+    it('adds an element to the back of the queue', function () {
+      queue.enqueue('dude')
+      queue.enqueue('bro')
+      queue.enqueue('foo')
+
+      expect(queue.back()).to.deep.equal('foo')
+    })
+
+    it('increases the queue size', function () {
+      queue.enqueue(10)
+      queue.enqueue(11)
+      queue.enqueue(12)
+
+      expect(queue.length()).to.equal(3)
+    })
+  })
+
+  context('dequeue()', function () {
+    it('returns and removes the front element in the queue', function() {
+      queue.enqueue(12)
+      expect(queue.dequeue()).to.deep.equal(12)
+    })
+
+    it('returns null if the queue is empty', function() {
+     expect(queue.dequeue()).to.deep.equal(null)
+    })
+  })
+
+  context('front()', function(){
+    it('returns the front element of the queue', function(){
+      queue.enqueue(10)
+      queue.enqueue(11)
+      queue.enqueue(12)
+      expect(queue.front()).to.deep.equal(10)
+    })
+
+    it('returns null if the queue is empty', function() {
+     expect(queue.front()).to.deep.equal(null)
+    })
+  })
+
+  context('back()', function(){
+    it('returns the back element of the queue', function(){
+      queue.enqueue(10)
+      queue.enqueue(11)
+      queue.enqueue(12)
+      expect(queue.back()).to.deep.equal(12)
+    })
+
+    it('returns null if the queue is empty', function() {
+     expect(queue.back()).to.deep.equal(null)
+    })
+  })
+
+  context('isEmpty()', function(){
+    it('returns true if queue is empty', function(){
+      queue.enqueue(10)
+      queue.enqueue(11)
+      queue.enqueue(12)
+      expect(queue.isEmpty()).to.equal(false)
+    })
+  })
+
+  context('length()', function(){
+    it('returns the number of elements in the queue', function(){
+      queue.enqueue(10)
```

Comment:
This is a totally acceptable approach to testing this.  I prefer a slightly more flexible approach that is less prone to human error:
```js
const expectedCount = 3

for( let counter = 0; counter < expectedCount; counter++ ) {
  queue.enqueue( counter )
}

expect( queue.length() ).to.equal( expectedCount )
```